### ISSUE TITLE
[rom,rom_ext] move first two attestation keygen stages (UDS, CDI_0) from ROM_EXT to ROM

### ISSUE THREAD

**Issue by timothytrippel** [lowRISC/opentitan#21583]
### Description

For DICE compliance, the UDS attestation key pair should be generated in ROM according to the diagram below. Additionally, since the UDS attestation key is derived from the ROM, and the CDI_0 key derived from the ROM_EXT, it is also (performance-wise) optimal to generate both key pairs in the ROM, since the ROM_EXT measurement is already taken in the ROM as part of the secure boot flow.

![OpenTitan DICE Key Ladder](https://github.com/lowRISC/opentitan/assets/5633066/57aa86ad-75c7-4f77-a211-4819aa02d725)

# Tasks
- [ ] move UDS attestation key/cert gen to the ROM 
- [ ] move CDI_0 attestation key/cert gen to the ROM 
- [ ] update website documentation with the above diagram

**Comment by timothytrippel** [comment]
Note: we considered moving UDS and CDI_0 key / cert gen to the ROM for a better security posture, but decided against it due to a hard 32k ROM size constraint (I.e., we ran out of room in the ROM to implement this). Moreover, performing singing operations (for the CDI_0 attestation certificate endorsement) was deemed potentially dangerous.

This issue has been updated to track the updated plan of record.

**Comment by johannheyszl** [comment]
Thanks @timothytrippel, I am removing the Hotlist label due to reached convergence on the topic. 

We have recorded security discussions for both variants in an (internal) [doc](https://docs.google.com/document/d/1zcXd0KnYZBRsNjPZE1x2V1fuSMjxQX9rxd7yyoYBx74/edit?usp=sharing&resourcekey=0-_mOEdXyu3E6cyLfH4tKj7Q) for later reference.

**Comment by jadephilipoom** [comment]
I just reviewed the doc today (I only heard of it yesterday) and as far as I can tell, the proposed solution still means that the ECDSA keygen/signing is fixed after manufacturing. To me, this raises the same concerns as before about side-channel vulnerabilities we can't patch. It buys only a few months of time to find and fix such issues, at the cost of more complex provisioning. This wasn't at all clear to me before reading the doc today, so I think it would be good to raise this in the security meeting one more time -- sorry, I know it's been a lot of rounds on this! I just think the new proposal is significantly different from previous discussions.

**Comment by timothytrippel** [comment]
Sorry @jadephilipoom, I may have been a bit unclear here. The plan is **_not_** to do any keygen/signing in the additional non-mutable ROM_EXT section for PROD. The plan is to do the keygen/signing in the ROM_EXT's mutable section as we originally discussed. The purpose of the non-mutable section is simply to provide an avenue for moving keygen/signing to what essentially is a ROM that is cheaper to update (it can be done at provisioning time; no need to re-spin a chip), in the future if we wanted to (only after gaining more confidence in the implementation of our algorithms.

Does that make sense?

**Comment by jadephilipoom** [comment]
Oh, I see. So to rephrase my understanding, this new section is not really going to be used for anything to start with, but provides us a way to more easily move stuff from ROM_EXT to ROM later if we gain more confidence and change our minds?

**Comment by timothytrippel** [comment]
yes, exactly.

**Comment by jadephilipoom** [comment]
Okay, that addresses my concern then, thanks! Sorry for the misunderstanding, I'll remove the hotlist label.

**Comment by jadephilipoom** [comment]
I took another look at the DICE spec and on this fourth pass I found some wording that makes me think there's no tension between the spec and signing in ROM_EXT actually. From [Hardware Requirements for Device Identifier Composition Engine](https://trustedcomputinggroup.org/wp-content/uploads/Hardware-Requirements-for-Device-Identifier-Composition-Engine-r78_For-Publication.pdf):

> The engine that performs the computation of the CDI may be updated, but those updates are not measured in the CDI and must be inherently trusted. First mutable code refers to the code that is executed after the Device Identifier Composition Engine and is not inherently trusted.

This sounds to me like in the spec, “first mutable code” does not actually mean first mutable code. It means the first code past “the DICE”, which is whatever code signs the first CDI cert and locks the UDS. The same document refers to “updatable DICE”:

> An updatable DICE SHALL only be updated through a secure process controlled by the manufacturer of the DICE.
> After updating through a manufacturer controlled process the DICE SHALL comply with this specification.
> The manufacturer controlled update process SHOULD NOT change the CDI.

So this sounds to me like what the spec calls “the DICE” can actually be mutable, as long as the manufacturer controls the process and guarantees that the code will continue to conform to DICE and won’t change the CDI. Since Silicon Creator signs ROM_EXT, and the only updates we’d expect to do are SCA fixes, I think this satisfies DICE requirements.

**Comment by johannheyszl** [comment]
nice - sounds we are all happy with 
- doing key gen and signing in the mutable ROM_EXT for now, and as we had planned originally (i.e. no change of plans), and
- having an option to include into a non-mutable ROM_EXT at a later stage if needed.

**Comment by timothytrippel** [comment]
The ROM_EXT non-mutable code section tasks have been added to a separate issue in #23425.

**Comment by moidx** [comment]
Moving to ROM_EXT as all ROM tasks tracked in this issue have been completed.
